### DISCUSSION TITLE
HCI spi driver implementation

### DISCUSSION THREAD

**Discussion by msmttchr** [zephyrproject-rtos/zephyr#66544] [Ideas]
As discussed in some PR #66227, #65172, there’s a proposal to split HCI over SPI implementation across different vendors files, since it is a bit tricky to keep several vendor specific implementation in the same driver.

I think there’s a general consensus about this, but we need also to evaluate whether make sense to:

- to keep a general common file and several vendor specific files 
- just have several vendor specific files 

if we select first option, is the drivers/bluetooth/hci/userchan.c a good existing option for the common file ?

I would like to know the opinion of @JordanYates, @HoZHel , @jhedberg, @aaronyegx or anyone else who would like to contribute

**Comment by aaronyegx** [comment]
Hi @msmttchr , thanks for creating this discussion. I think it would be better to have one common file for HCI SPI then the vendors can extend their specific feature only in vendor specific files. But before doing this all the vendors using the HCI SPI currently should align that they will refactor the current driver to work with new created common files, then the job could make much sense. It may need some big effort because all the vendors need to retest the functions and performance in their platforms.
